AEMG/training.py

```diff
@@ -143,11 +146,10 @@ def train(self, epochs=1000, patience=50, weight=[1,1,1,0]):
                 epoch_train_loss += loss_total.item()
 
             if weight[3] != 0:
-                for i, (x_final, label) in enumerate(self.labels_loader):
+                for i, (triplets, x_final) in enumerate(self.labels_loader):
                     x_final = x_final.to(self.device)
-                    label = label.to(self.device)
                     z_final = self.encoder(x_final)
-                    loss_con = self.labels_losses(z_final, label, weight)
+                    loss_con = self.labels_losses(z_final, triplets)
```

Comment:
We would still need the `weight` to be applied to `loss_con` before `optimizer.step()`

Reply:
But isn't that already happening in line 153?